### ISSUE TITLE
Schould tests in this repo run on a cron job?

### ISSUE THREAD

**Issue by rgommers** [data-apis/dataframe-interchange-tests#12]
It would probably be nice to have these tests run once a week, in case a new release of a dataframe library happens. Not urgent, but perhaps useful. It should be a matter of adding something like this to the existing `.github/workflows/test.yml`:
```yaml
on:
  schedule:
  #        ┌───────────── minute (0 - 59)
  #        │  ┌───────────── hour (0 - 23)
  #        │  │ ┌───────────── day of the month (1 - 31)
  #        │  │ │ ┌───────────── month (1 - 12 or JAN-DEC)
  #        │  │ │ │ ┌───────────── day of the week (0 - 6 or SUN-SAT)
  #        │  │ │ │ │
  - cron: "9  9 * * 6"
```

**Comment by honno** [comment]
Yep I can have a go at some point, after I merge #11*.

\* was going to wait on vaex but actually now that we have up-to-date implementations in pandas and modin (+cudf, but can't run that on CI), I can get #11 merged now after xfail tweaks.

**Comment by jorisvandenbossche** [comment]
This would be good to have (also to allow you to see recent test logs)

**Comment by jorisvandenbossche** [comment]
I find it also suspicious that the tests actually passed in September. Anything converting a dataframe with column of strings to pandas (eg vaex to pandas) should have failed until recently (https://github.com/pandas-dev/pandas/pull/50565). And it seems that string column should normally be covered in the roundtrip tests.

**Comment by honno** [comment]
> This would be good to have (also to allow you to see recent test logs)

My biggest de-motivator here was that [these nightly pandas builds](https://anaconda.org/scipy-wheels-nightly/pandas) suddenly weren't actually nightly. I spent some time exploring this to no avail, will have another go though.

> Anything converting a dataframe with column of strings to pandas (eg vaex to pandas) should have failed until recently ([pandas-dev/pandas#50565](https://github.com/pandas-dev/pandas/pull/50565)). And it seems that string column should normally be covered in the roundtrip tests.

There are quite a few xfails (and skips when flaky), including vaex-to-pandas. I could probably highlight this in the README.

https://github.com/data-apis/dataframe-interchange-tests/blob/8c1fc08716629ad4bcbb089b2a961da13c0e234c/tests/conftest.py#L41-L78

Also generally some dtypes are disabled as they're known to be erroneous in [`wrrapers.py`](https://github.com/data-apis/dataframe-interchange-tests/blob/main/tests/wrappers.py), e.g. 
datatimes and strings were disabled for cuDF

https://github.com/data-apis/dataframe-interchange-tests/blob/8c1fc08716629ad4bcbb089b2a961da13c0e234c/tests/wrappers.py#L328-L333

**Comment by jorisvandenbossche** [comment]
> There are quite a few xfails (and skips when flaky), including vaex-to-pandas.

Hmm, so basically since there is only a single roundtrip test, the full roundtrip tests are skipped for many combinations. I suppose ideally we would only skip certain data types in the roundtrip, given the exact orig/dest library, but still run the test itself? 
I suppose that is already the idea of `supported_dtypes`, and ideally the vaex tests would be unskipped but maybe their supported_dtypes limited to get it passing?

**Comment by honno** [comment]
> > There are quite a few xfails (and skips when flaky), including vaex-to-pandas.
> 
> I suppose ideally we would only skip certain data types in the roundtrip, given the exact orig/dest library, but still run the test itself? I suppose that is already the idea of `supported_dtypes`

Yep!

> and ideally the vaex tests would be unskipped but maybe their supported_dtypes limited to get it passing?

vaex has a [few bugs](https://github.com/vaexio/vaex/issues/created_by/honno)) which tests (inc. the roundtrips) from passing beyond just a lack of dtype support. I think that goes the same for the other adopters, but yeah definitely need to revisit this + add a cron job.
